with-pgbadger

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash -eu
+set -o pipefail
+
+log() {
+  echo "[with-pgbadger] $*"
+}
+
+badger_date() {
+  date '+%Y-%m-%d %H:%M:%S'
+}
+
+if [[ -z "${POSTGRES_LOG_FILE-}" ]]; then
+  cat <<EOF
+
+  !!! POSTGRES_LOG_FILE NOT SET!
+  !!!
+  !!! pgbadger needs to know where your log file is in order to process it
+  !!!
+  !!! try again e.g.:
+  !!!
+  !!!     POSTGRES_LOG_FILE=/var/log/postgresql/postgresql-12-main.log $0 $*
+
+EOF
+  exit 1
+fi
+
+reportsDir='./.pgbadger'
+mkdir -p "$reportsDir"
+
+start="$(badger_date)"
+
+log "Enabling required postgres options..."
+# see: https://github.com/darold/pgbadger#postgresql-configuration
+psql >/dev/null -c 'ALTER SYSTEM SET log_min_duration_statement = 0'
```

Comment:
Add `ALTER SYSTEM SET log_line_prefix = '%m [%p] '` to guarantee that there is a log prefix.

Reply:
It'd be nice to have an undo on this because the logs get heavy. On my default Postgres 9.6 install, these are the defaults...

```sql
ALTER SYSTEM SET log_checkpoints = off;
ALTER SYSTEM SET log_connections = off;
ALTER SYSTEM SET log_disconnections = off;
ALTER SYSTEM SET log_lock_waits = off;
ALTER SYSTEM SET log_temp_files = -1;
ALTER SYSTEM SET log_autovacuum_min_duration = -1;
ALTER SYSTEM SET log_error_verbosity = default;
ALTER SYSTEM SET log_duration = off;
ALTER SYSTEM SET log_statement = 'none';
ALTER SYSTEM SET log_line_prefix = '';
```

Reply:
To do reliably, this should read the initial values for each setting and then restore these at the end of the script, regardless of success.  Care should be take to handle failure within either reading the initial settings, or restoring.

@yanokwa would it be acceptable to just communicate that settings may change, and how to restore them, as part of the script?

Reply:
Is it easy to say settings X,Y,Z with values A,B,C will change and ask for confirmation to continue? Then add a force flag for those who want to do this in CI.